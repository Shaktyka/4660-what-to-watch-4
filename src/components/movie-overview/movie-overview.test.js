import React from 'react';
import renderer from 'react-test-renderer';

import MovieOverview from './movie-overview.jsx';

const filmData = {
  ratingScore: 6.7,
  ratingCount: 410,
  description: [
    `Mindhunter`,
    `The film`,
    `They reach`
  ],
  director: `Bryan Singer`,
  starring: [`Rami Malek`, `Lucy Boynton`, `Gwilym Lee`]
};

describe(`MovieOverview component rendering`, () => {

  it(`MovieOverview renders correctly`, () => {
    const {ratingScore, ratingCount, description, director, starring} = filmData;

    const tree = renderer
      .create(
          <MovieOverview
            ratingScore={ratingScore}
            ratingCount={ratingCount}
            description={description}
            director={director}
            starring={starring}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

});
