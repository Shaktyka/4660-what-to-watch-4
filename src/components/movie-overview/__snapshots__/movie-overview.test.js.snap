// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieOverview component rendering MovieOverview renders correctly 1`] = `
Array [
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      6.7
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Good
      </span>
      <span
        className="movie-rating__count"
      >
        410
         ratings
      </span>
    </p>
  </div>,
  <div
    className="movie-card__text"
  >
    <p>
      Mindhunter
       
    </p>
    <p>
      The film
       
    </p>
    <p>
      They reach
       
    </p>
    <p
      className="movie-card__director"
    >
      <strong>
        Director: 
        Bryan Singer
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: 
        Rami Malek, Lucy Boynton, Gwilym Lee
      </strong>
    </p>
  </div>,
]
`;
